include/gevolution/Particles_gevolution.hpp

```diff
@@ -94,16 +99,211 @@ class Particles_gevolution :
         particle_dataType>
 {
   public:
+    
     using value_type = particle; // helper attribute used in containers for metaprogramming
 
     void saveGadget2 (std::string filename, gadget2_header &hdr,
                       const int tracer_factor = 1) const ;
-    void saveGadget2 (std::string filename, gadget2_header &hdr,
-                      lightcone_geometry &lightcone, double dist, double dtau,
-                      double dtau_old, double dadtau,
-                      double vertex[MAX_INTERSECTS][3], const int vertexcount,
-                      std::set<long> &IDbacklog, std::set<long> &IDprelog,
-                      Field<Real> *phi, const int tracer_factor = 1);
+   
+    template<typename selector_type>
+    void saveGadget2 (
+        const std::string filename, 
+        gadget2_header hdr,
+        selector_type select_function,
+        int const tracer_factor = 1)const
+    // precondition: hdr must already contain cosmological data and number of particles=0
+    // FIXME: convert units from gevolution to gadget2
+    // FIXME: convert momentum to velocities. is this required at all?
```

Comment:
Still is not clear to me if we need velocities or momentum of particles in the lightcone. My guess is momentum. But we need to take a look at: 1) the legacy code 2) the observables (eg. redshifts) and how are they computed from particle properties.

Reply:
In the legacy code Julian saved momentum/mass/a as gadget vel. That would be velocity dx/dtau (comoving distance per unit of conformal time) in the case of an homogeneous universe and non-relativistic particle. I will keep the same convention, but I have a feeling that the observables in a relativistic setup should be computed from covariant quantities, like momentum.